opentelemetry-otlp/src/metric.rs

```diff
@@ -61,14 +81,13 @@ where
 /// Note that currently the OTLP metrics exporter only supports tonic as it's grpc layer and tokio as
 /// runtime.
 #[derive(Debug)]
-pub struct OtlpMetricPipelineBuilder<AS, ES, SP, SO, I, IO>
+pub struct OtlpMetricPipelineBuilder<ES, SP, SO, I, IO>
 where
-    AS: AggregatorSelector + Send + Sync + 'static,
     ES: ExportKindFor + Send + Sync + Clone + 'static,
     SP: Fn(PushControllerWorker) -> SO,
     I: Fn(time::Duration) -> IO,
 {
-    aggregator_selector: AS,
+    aggregator_selector: Box<dyn AggregatorSelector + Send + Sync + 'static>,
```

Comment:
Are all these other box changes still needed?

Reply:
I think so. I tried it before and was running into problems. If my understanding is correct if you remove Box::dyn then the compiler does static dispatch and turns T into a concrete type. So without dyns, it works in a situation where both ```opentelemetry_otlp::new_metrics_pipeline(tokio::spawn, delayed_interval)``` and ```.with_aggregator_selector()``` use the same type T.

Reply:
If you replace the method body with the full:

```rust
pub fn with_aggregator_selector<T>(
    self,
    aggregator_selector: T,
) -> OtlpMetricPipelineBuilder<T, ES, SP, SO, I, IO>
where
    T: AggregatorSelector + Send + Sync + 'static,
{
    OtlpMetricPipelineBuilder {
        aggregator_selector,
        export_selector: self.export_selector,
        spawn: self.spawn,
        interval: self.interval,
        export_config: self.export_config,
        tonic_config: self.tonic_config,
        resource: self.resource,
        stateful: self.stateful,
        period: self.period,
        timeout: self.timeout,
    }
}
```

Then you don't need any other change.

Reply:
I have a feeling I am missing something here.
If I use the snippet from above I am getting compilation errors on both the main and this branch. 
If I remove ```Box::dyns``` I am pretty much back to the code that is currently on the main branch. In there the aggregator selector in  ```new_metrics_pipeline``` must be of the same type as in ```with_aggreagtor_selector```.

Reply:
i have removed more dyns from ```new_metrics_pipeline_with_selector``` but i am not sure if i can remove the remaining ones

Reply:
Something along the lines of https://github.com/open-telemetry/opentelemetry-rust/commit/88b1904b808ae811076eddc53a4c87e01181999a should work

Reply:
@jtescher I have missed that. The code is much simpler indeed, thanks. I have merged your suggestions, so hopefully, we are good now.